src/core/api/border_routing_api.cpp

```diff
@@ -194,6 +194,28 @@ otError otBorderRoutingGetNextRouterEntry(otInstance                         *aI
     return AsCoreType(aInstance).Get<BorderRouter::RoutingManager>().GetNextRouterEntry(*aIterator, *aEntry);
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_TRACK_PEER_BR_INFO_ENABLE
+
+otError otBorderRoutingGetNextPeerBrEntry(otInstance                           *aInstance,
+                                          otBorderRoutingPrefixTableIterator   *aIterator,
+                                          otBorderRoutingPeerBorderRouterEntry *aEntry)
+{
+    AssertPointerIsNotNull(aIterator);
+    AssertPointerIsNotNull(aEntry);
+
+    return AsCoreType(aInstance).Get<BorderRouter::RoutingManager>().GetNextPeerBrEntry(*aIterator, *aEntry);
+}
+
+uint16_t otBorderRoutingCountPeerBrs(otInstance *aInstance, uint32_t *aMinAge)
```

Comment:
Hi @abtink , just want to understand what is the consideration here to make it as a BorderRouting API?

can we make it as a network data API, so that the devices without border routing enabled can also have ability to know how many peer BRs are there in the network data?

Reply:
The main reason is to not only track the RLOC16 of peer BRs but also their age (how long they've been present in the network data). In the future, we may learn and track other information per peer BR.

Restricting this to when the device itself is also a BR (is enabled for `BorderRouting`) ensures that the device has sufficient resources (computational power and heap/memory) for this task.

The age can be helpful in many situations. It helps us avoid making decisions based on transient changes (e.g., a BR rebooting and adding/removing its entries in network data, a BR attaching as a child and then getting promoted to a router).

There are other ideas/enhancements that would utilize the list of peer BRs, such as synchronizing the "address cache table" among BRs, using "netdiag" to learn more about a peer BR (ext address, version, vendor info, etc), or implementing new mechanisms to detect if a peer BR goes offline without removing its entries in Network Data (BRs policing each other (https://github.com/openthread/openthread/issues/9591)).

----

> can we make it as a network data API, so that the devices without border routing enabled can also have ability to know how many peer BRs are there in the network data?

We can always iterate over all entries in Network Data and learn of all BRs (using existing `otNetdata` APIs). Or  expose the `NetworkData::CountBorderRouters()` as an `otNetData` API for this, but generally, I prefer to have one API for this purpose.

Reply:
Thanks @abtink ,  I am asking this because I am trying to [use this API to upload the peer BR count ](https://github.com/openthread/ot-br-posix/pull/2383/files#diff-6c432a543957f971d04ffaf4ed54864c00b790f8e1c79cb225a77cc0baa11414R1238)as a new telemetry metrics but found I can only use this when `OTBR_ENABLE_BORDER_ROUTING` is true.

Thanks for your explanation! I agree that having only one API for this purpose, and it's reasonable that we only enable this API on a BR device considering the more resources this feature can consume.